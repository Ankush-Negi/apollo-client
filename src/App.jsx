
import React from 'react';
import { ApolloProvider } from '@apollo/react-components';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';
import {
  BrowserRouter as Router, Route, Redirect, Switch,
} from 'react-router-dom';
import { theme } from './theme';
import {
  TextFieldDemo, ChildrenDemo, TraineeComponent, Wrapper, InputDemo, NoMatch,
} from './pages';
import { AuthLayoutRoute, PrivateLayoutRoute } from './routes/index';
import { SnackBarProvider } from './contexts';
import client from './lib/apollo-client';


function App() {
  return (
    <SnackBarProvider>
      <ApolloProvider client={client}>
        <MuiThemeProvider theme={theme}>
          <CssBaseline />
          <Router>
            <Switch>
              <Route exact path="/">
                <Redirect to="/login" />
              </Route>
              <AuthLayoutRoute exact path="/login" component={Wrapper} />
              <PrivateLayoutRoute exact path="/trainee" component={TraineeComponent} />
              <PrivateLayoutRoute exact path="/text-field-demo" component={TextFieldDemo} />
              <PrivateLayoutRoute exact path="/children-demo" component={ChildrenDemo} />
              <PrivateLayoutRoute exact path="/input-demo" component={InputDemo} />
              <PrivateLayoutRoute component={NoMatch} />
            </Switch>
          </Router>
        </MuiThemeProvider>
      </ApolloProvider>
    </SnackBarProvider>

  );
}

export default App;
