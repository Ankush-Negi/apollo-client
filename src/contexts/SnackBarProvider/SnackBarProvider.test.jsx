import React from 'react';
import { mount } from 'enzyme';
import App from '../../App';
import { MyContext, SnackBarProvider } from './SnackBarProvider';

const mockProps = {
  open: 'true',
  onClose: 'close',
  status: 'running',
  message: 'All Good',
  children: { App },
};

describe('SnackBar Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <>
        <MyContext.Provider>
          <SnackBarProvider {...mockProps} />
        </MyContext.Provider>
      </>,
    );
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('snackbar length', () => {
    const opened = wrapper.find('SnackBars');
    expect(opened).toHaveLength(0);
  });
  it('snackbar value', () => {
    const opened = wrapper.find(SnackBarProvider).value;
    expect(opened).toEqual(undefined);
  });
  it('snackbar children', () => {
    const opened = wrapper.find(MyContext.Provider).value;
    expect(opened).toBe(undefined);
  });
});
