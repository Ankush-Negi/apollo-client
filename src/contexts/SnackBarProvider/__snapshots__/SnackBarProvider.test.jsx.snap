// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackBar Component should render correctly 1`] = `
<SnackBarProvider
  message="All Good"
  onClose="close"
  open="true"
  status="running"
>
  <Snackbars
    message=""
    onClose={[Function]}
    open={false}
    status=""
  >
    <div>
      <WithStyles(ForwardRef(Snackbar))
        autoHideDuration={5000}
        onClose={[Function]}
        open={false}
      >
        <ForwardRef(Snackbar)
          autoHideDuration={5000}
          classes={
            Object {
              "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter",
              "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft",
              "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight",
              "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter",
              "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft",
              "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight",
              "root": "MuiSnackbar-root",
            }
          }
          onClose={[Function]}
          open={false}
        />
      </WithStyles(ForwardRef(Snackbar))>
    </div>
  </Snackbars>
</SnackBarProvider>
`;
